frontend/Dockerfile

```
@@ -0,0 +1,22 @@
+FROM node:18 AS base
+
+WORKDIR /frontend
+
+COPY package.json .
+
+RUN npm install
+
+COPY src src
+COPY build build
+COPY public public
+COPY tailwind.config.ts .
```

Not saying strictly better, but worth noting we can combine these since you're not changing any names:
```suggestion
COPY src build public tailwind.config.ts ./
```